tests/30rooms/12thirdpartyinvite.pl

```diff
@@ -163,6 +163,54 @@ sub can_invite_unbound_3pid
       })->followed_by( assert_membership( "join" ) );
    };
 
+test "Can accept third party invite with /join",
+   requires => [ local_user_fixture(), local_user_fixture(),
+                 $main::HOMESERVER_INFO[1], id_server_fixture() ],
+
+   do => sub {
+      my ( $inviter, $invitee, $info, $id_server ) = @_;
+      my $hs_uribase = $info->client_location;
+
+      my $room_id;
+
+      matrix_create_room( $inviter, visibility => "private" )
+      ->then( sub {
+         ( $room_id ) = @_;
+
+         do_3pid_invite( $inviter, $room_id, $id_server->name, $invitee_email )
+      })->then( sub {
+         matrix_get_room_state( $inviter, $room_id, )
+      })->then( sub {
+         my ( $body ) = @_;
+
+         my $token;
+
+         foreach my $event (@$body) {
```

Comment:
I'd use

    my $invite_event = first { $_->{type} eq "m.room.third_party_invite" } @$body or
       die "Could not find 3rd-party invite event";

    my $token = $event->{state_key};